test/e2e/generate/generate_test.go

```diff
@@ -0,0 +1,288 @@
+	}
+	// Generate E2E Client ==================
+	e2eClient, err := NewE2EClient()
+	Expect(err).To(BeNil())
+	// ======================================
+
+	// ====== Range Over ClusterRoleTests ==================
+	for _, tests := range ClusterRoleTests {
+		By(fmt.Sprintf("Test to generate ClusterRole and ClusterRoleBinding : %s", tests.TestName))
+		By(fmt.Sprintf("synchronize = %v\t clone = %v", tests.Sync, tests.Clone))
+
+		// ======= CleanUp Resources =====
+		By(fmt.Sprintf("Cleaning Cluster Policies from Namespace : %s", clPolNS))
+		e2eClient.CleanClusterPolicies(clPolGVR, clPolNS)
+		// Clear Namespace
+		By(fmt.Sprintf("Deleting Namespace : %s\n", tests.ResourceNamespace))
+		e2eClient.DeleteClusteredResource(nsGVR, tests.ResourceNamespace)
+
+		// Wait Till Deletion of Namespace
+		GetWithRetry(time.Duration(1), 15, func() error {
+			_, err := e2eClient.GetClusteredResource(nsGVR, tests.ResourceNamespace)
+			if err != nil {
+				return nil
+			}
+			return errors.New("Deleting Namespace")
+		})
+
+		// If Clone is true Clear Source Resource and Recreate
+		if tests.Clone {
+			By(fmt.Sprintf("Clone = true, Deleting Source ClusterRole and ClusterRoleBinding from Clone Namespace : %s\n", tests.CloneNamespace))
+			// Delete ClusterRole to be cloned
+			e2eClient.DeleteNamespacedResource(crGVR, tests.CloneNamespace, tests.ClonerClusterRoleName)
+			// Delete ClusterRoleBinding to be cloned
+			e2eClient.DeleteNamespacedResource(crbGVR, tests.CloneNamespace, tests.ClonerClusterRoleBindingName)
+		}
+
+		// Wait to Delete Resources
+		// time.Sleep(10 * time.Second)
+		// ====================================
+
+		// ======== Create ClusterRole Policy =============
+		By(fmt.Sprintf("Creating Generate Role Policy in %s", clPolNS))
+		_, err = e2eClient.CreateNamespacedResourceYaml(clPolGVR, clPolNS, tests.Data)
+		Expect(err).NotTo(HaveOccurred())
+		// ============================================
+
+		// ======= Create Namespace ==================
+		By(fmt.Sprintf("Creating Namespace which triggers generate %s \n", clPolNS))
+		_, err = e2eClient.CreateClusteredResourceYaml(nsGVR, namespaceYaml)
+		Expect(err).NotTo(HaveOccurred())
+
+		// Wait Till Creation of Namespace
+		GetWithRetry(time.Duration(1), 15, func() error {
+			_, err := e2eClient.GetClusteredResource(nsGVR, tests.ResourceNamespace)
+			if err != nil {
+				return err
+			}
+			return nil
+		})
+		// ===========================================
+
+		// If Clone is true Create Source Resources
+		if tests.Clone {
+			By(fmt.Sprintf("Clone = true, Creating Cloner Resources in Namespace : %s", tests.CloneNamespace))
+			// Create ClusterRole to be cloned
+			_, err := e2eClient.CreateClusteredResourceYaml(crGVR, tests.CloneSourceClusterRoleData)
+			Expect(err).NotTo(HaveOccurred())
+			// Create ClusterRoleBinding to be cloned
+			_, err = e2eClient.CreateClusteredResourceYaml(crbGVR, tests.CloneSourceClusterRoleBindingData)
+			Expect(err).NotTo(HaveOccurred())
+		}
+
```

Comment:
The order seemed confused to me: we first create trigger namespace, then create the resources that are to be cloned? Won't this cause error?

Reply:
Yes, It will cause issues and I'll move it before creating Namespace trigger.